--- alsaplayer-0.99.71/Makefile.am.orig	Mon Jun 10 22:51:09 2002
+++ alsaplayer-0.99.71/Makefile.am	Wed Jul 10 05:25:55 2002
@@ -1,5 +1,5 @@
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = intl po interface output input app alsaplayer extra docs \
+SUBDIRS = intl po interface output input app alsaplayer extra \
 	libalsaplayer examples m4 reader scopes2
 
 EXTRA_DIST = config.rpath bootstrap alsaplayer.spec
